### PR TITLE
system76-oled: init at 0.1.4

### PR DESCRIPTION
**Pull request by zarthross**
###### Motivation for this change

System76 Adder WS line of laptops use an OLED display. The standard brightness controls do not work with these OLED screens.

This package provides the System76-oled application that System76 uses with PopOS to provide brightness controls on these notebooks.  

This was tested on a System76 Adder WS 1 (addw1) 

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/system76-oled/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  buildInputs = [ dbus libX11 libXrandr ];
+
+  cargoBuildFlags = [ ];
+
+  cargoSha256 = "sha256:13nhs17gzya1by5b301hl8q0pm9m9k1pnzmmgx1v1qam0zd7qj91";
+
+  installPhase = ''
+    install -D -m -0644 data/system76-oled.desktop $out/etc/xdg/autostart/system76-oled.desktop
+    install -D -m 0755 target/*/release/system76-oled $out/bin/system76-oled
+  '';
```
Comment (legendofmiracles) [inline pkgs/os-specific/linux/system76-oled/default.nix:25]:
```suggestion
  postInstall = ''
    install -D -m -0644 data/system76-oled.desktop $out/etc/xdg/autostart/system76-oled.desktop
  '';
```
The normal rust hooks already take care of copying the binary for you

File: pkgs/os-specific/linux/system76-oled/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  version = "0.1.4";
+
+  src = fetchFromGitHub {
+    owner = "pop-os";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256:0xy6fpr9k5wnq074flsxm1k27svvk4pvqyryhdsw1jpch46lzm1f";
+  };
+
+  nativeBuildInputs = [ pkg-config makeWrapper ];
```
Comment (legendofmiracles) [inline pkgs/os-specific/linux/system76-oled/default.nix:14]:
```suggestion
  nativeBuildInputs = [ pkg-config ];
```
Not used

File: pkgs/os-specific/linux/system76-oled/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  buildInputs = [ dbus libX11 libXrandr ];
+
+  cargoBuildFlags = [ ];
```
Comment (legendofmiracles) [inline pkgs/os-specific/linux/system76-oled/default.nix:18]: Is this not the default?
```suggestion
```

File: pkgs/os-specific/linux/system76-oled/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
+{ rustPlatform, lib, fetchFromGitHub, pkg-config, makeWrapper, dbus, libX11
+, libXrandr }:
+rustPlatform.buildRustPackage rec {
```
Comment (legendofmiracles) [inline pkgs/os-specific/linux/system76-oled/default.nix:3]:
```suggestion
{ lib
, rustPlatform
, lib
, fetchFromGitHub
, pkg-config
, makeWrapper
, dbus
, libX11
, libXrandr
}:

rustPlatform.buildRustPackage rec {
```
Makes the diff nicer, and for derivations with rather more arguments this is preferred

File: pkgs/os-specific/linux/system76-oled/default.nix (hunk 1 above)
Comment (zarthross) [reply pkgs/os-specific/linux/system76-oled/default.nix:30]: For whatever reason, it does not appear to be the case:
```
[nix-shell:~/nixpkgs]$ nix-build -KA system76-oled
/nix/store/vqmf7b1j319pbjpnad0gzyy80c99b75g-system76-oled-0.1.4

[nix-shell:~/nixpkgs]$ ls -R result
result:
etc

result/etc:
xdg

result/etc/xdg:
autostart

result/etc/xdg/autostart:
system76-oled.desktop

[nix-shell:~/nixpkgs]$ 
```

**Comment by zarthross** [comment]
@legendofmiracles Thanks for the feedback!  Greatly appreciated!

**Comment by zarthross** [comment]
This utility appears to not be necessary on Linux Kernel 5.15.18.  So I'm closing this PR as unneeded.

**Pull request closed without merging**